### ISSUE TITLE
Periodic Errors

### ISSUE THREAD

**Issue by doug-wade** [zeit/hyperterm#542]
I installed Hyperterm from the [download link](https://hyperterm-updates.now.sh/download/osx) on Saturday, and I've been getting this error periodically

```
Uncaught Exception:
Error: watch /Users/doug.wade/.node-spawn-wrap-70179-0911932639cb ENOENT
    at exports._errnoException (util.js:949:11)
    at FSWatcher.start (fs.js:1433:19)
    at Object.fs.watch (fs.js:1460:11)
    at Gaze._watchDir (/Applications/HyperTerm.app/Contents/Resources/app/node_modules/gaze/lib/gaze.js:288:30)
    at /Applications/HyperTerm.app/Contents/Resources/app/node_modules/gaze/lib/gaze.js:357:10
    at iterate (/Applications/HyperTerm.app/Contents/Resources/app/node_modules/gaze/lib/helper.js:69:5)
    at /Applications/HyperTerm.app/Contents/Resources/app/node_modules/gaze/lib/helper.js:78:11
    at /Applications/HyperTerm.app/Contents/Resources/app/node_modules/gaze/lib/gaze.js:422:5
    at iterate (/Applications/HyperTerm.app/Contents/Resources/app/node_modules/gaze/lib/helper.js:69:5)
    at Object.forEachSeries (/Applications/HyperTerm.app/Contents/Resources/app/node_modules/gaze/lib/helper.js:83:3)
```

It seems correlated with `npm` use (i.e. I'm more likely to get the error when `npm` is running, either in hyperterm or in iTerm2), though I'm not certain about that.  There is, in fact, no .node-spawn-wrap when I check now, after dismissing the error

```
dougwade redfin.npm/customer-ui ‹dbw-unit-test-customer-ui› » ls /Users/doug.wade/.node-spawn*                                                                                1 ↵
zsh: no matches found: /Users/doug.wade/.node-spawn*
```

I'm running on node 6.3.1 with OS X 10.11.6.  I got it today by starting hyperterm and leaving it run for ten or so minutes, without interacting with the terminal at all, so I believe its either in my setup or hyperterm has a 🐛 ; I'm leaning towards the second, since the only configuration I've done is add the hyperterm-atom-dark plugin

```
module.exports = {
  config: {
    // default font size in pixels for all tabs
    fontSize: 12,

    // font family with optional fallbacks
    fontFamily: 'Menlo, "DejaVu Sans Mono", "Lucida Console", monospace',

    // terminal cursor background color and opacity (hex, rgb, hsl, hsv, hwb or cmyk)
    cursorColor: 'rgba(248,28,229,0.75)',

    // `BEAM` for |, `UNDERLINE` for _, `BLOCK` for █
    cursorShape: 'BLOCK',

    // color of the text
    foregroundColor: '#fff',

    // terminal background color
    backgroundColor: '#000',

    // border color (window, tabs)
    borderColor: '#333',

    // custom css to embed in the main window
    css: '',

    // custom css to embed in the terminal window
    termCSS: '',

    // custom padding (css format, i.e.: `top right bottom left`)
    padding: '12px 14px',

    // the full list. if you're going to provide the full color palette,
    // including the 6 x 6 color cubes and the grayscale map, just provide
    // an array here instead of a color map object
    colors: {
      black: '#000000',
      red: '#ff0000',
      green: '#33ff00',
      yellow: '#ffff00',
      blue: '#0066ff',
      magenta: '#cc00ff',
      cyan: '#00ffff',
      white: '#d0d0d0',
      lightBlack: '#808080',
      lightRed: '#ff0000',
      lightGreen: '#33ff00',
      lightYellow: '#ffff00',
      lightBlue: '#0066ff',
      lightMagenta: '#cc00ff',
      lightCyan: '#00ffff',
      lightWhite: '#ffffff'
    },

    // the shell to run when spawning a new session (i.e. /usr/local/bin/fish)
    // if left empty, your system's login shell will be used by default
    shell: ''

    // for advanced config flags please refer to https://hyperterm.org/#cfg
  },

  // a list of plugins to fetch and install from npm
  // format: [@org/]project[#version]
  // examples:
  //   `hyperpower`
  //   `@company/project`
  //   `project#1.0.1`
  plugins: ['hyperterm-atom-dark'],

  // in development, you can create a directory under
  // `~/.hyperterm_plugins/local/` and include it here
  // to load it and avoid it being `npm install`ed
  localPlugins: []
};
```

**Comment by rauchg** [comment]
It seems like it's related to `gaze` watching of our config file. Not sure why it'd error, but I suspect we're better off suppressing it *provided* that the file continues to be watched. If someone gets this error, it'd be interesting to see if subsequent changes still gets captured.

Otherwise, @shama have you run into this?

**Comment by shama** [comment]
That error should be fixed by https://github.com/shama/gaze/commit/264101644d105f010e382e371db4eda0e45cbdfd and going out in `1.1.1` as soon as the Windows tests pass.

**Comment by MrRio** [comment]
@shama great thanks shama :)

**Comment by rauchg** [comment]
wow @shama thanks <3

**Comment by shama** [comment]
Published as `1.1.1`

**Comment by graemec23** [comment]
I get this error whenever I try and run Hyperterm and Atom at the same time

**Comment by whitelynx** [comment]
Actually, it looks like this is a duplicate of #444. (or at least related)

**Comment by doug-wade** [comment]
@whitelynx I think #444 is different, since its an ENOENT on `/user/local/bin/fish`, which is outside the scope of hyperterm.  My error, the ENOENT on `/Users/doug.wade/.node-spawn-wrap-70179-0911932639cb` is a file that is created and managed by hyperterm (or possibly atom and mistakenly monitored by hyperterm per @graemec23's discovery that this only happens with Atom open, which _seems_ consistent with my experience).

@MrRio I noticed you've marked this as "Awaiting Response" -- I've installed the new version, and the error still repros.  Is there _more_ response that you're awaiting?

**Comment by graemec23** [comment]
@doug-wade I'm not long receiving errors having Atom and Hyperterm open at the same time, are you using latest version of Atom?

**Comment by doug-wade** [comment]
@graemec23 I'm on Atom 1.9.9 -- upgrading does seem to help, but I'm still getting the error.  I got it twice so far today, both times when running and `npm install && npm update`

<img width="431" alt="screen shot 2016-08-29 at 12 53 49 pm" src="https://cloud.githubusercontent.com/assets/3477707/18065264/122c45d4-6de8-11e6-9fb5-00166db865a8.png">

**Comment by graemec23** [comment]
@doug-wade Have you tried installing hyperterm with homebrew?

**Comment by doug-wade** [comment]
@graemec23 I had it installed both from the site and from homebrew.  I uninstalled the non-homebrew one, and I'm still getting the error. For reference, I've been running HyperTerm 0.7.1.36, Atom 1.9.9 and npm 2.14.12 concurrently every time I've gotten these errors in the past 2-3 days.

**Comment by matheuss** [comment]
Closing in favor of #225

**Comment by whitelynx** [comment]
@doug-wade said:
> @whitelynx I think #444 is different, since its an ENOENT on `/user/local/bin/fish`, which is outside the scope of hyperterm. My error, the ENOENT on `/Users/doug.wade/.node-spawn-wrap-70179-0911932639cb` is a file that is created and managed by hyperterm (or possibly atom and mistakenly monitored by hyperterm per @graemec23's discovery that this only happens with Atom open, which seems consistent with my experience).

Actually, the original issue reported in #444 is an `ENOENT` on `/Users/iradchenko/.google-cloud-sdk.staging`; the `/usr/local/bin/fish` thing was someone else's error due to misconfiguration. At any rate, this one has been closed in favor of #225, but the similarity with #444 would point to this not being directly related to files HyperTerm is creating, but rather ones that it's watching, but shouldn't. (I'm pretty sure `/Users/iradchenko/.google-cloud-sdk.staging` isn't created by HyperTerm, at least)
